### PR TITLE
Move to GitLink 3.0

### PR DESCRIPTION
**Pull request by jaredpar**
This makes two changes:

1. Move us to GitLink 3.0
1. Add source link info to all binaries we sign

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-ide

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure 

Sorry IDE, that was a typo

**Comment by jaredpar** [comment]
retest windows_debug_unit32_prtest please

File: build/scripts/build-utils.ps1 [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Set-StrictMode -version 2.0
+$ErrorActionPreference="Stop"
+
+# Declare a number of useful variables for other scripts to use
+[string]$repoDir = Resolve-Path (Join-Path $PSScriptRoot "..\..")
+[string]$binariesDir = Join-Path $repoDir "Binaries"
+[string]$scriptDir = $PSScriptRoot
+
+function Create-Directory([string]$dir) {
+    New-Item $dir -ItemType Directory -ErrorAction SilentlyContinue | out-null
```
Comment (jasonmalinowski) [inline build/scripts/build-utils.ps1]: What happens if it legitimately fails to create the directory?

File: src/Tools/MicroBuild/run-gitlink.ps1 [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Run GitLink on the binaries that we are producing.
```
Comment (jasonmalinowski) [inline src/Tools/MicroBuild/run-gitlink.ps1]: Can you include a comment as to why the old mechanism wasn't sufficient?

Comment (jasonmalinowski) [inline src/Tools/MicroBuild/run-gitlink.ps1]: (if the answer was "it's no longer supported", then that should have been written in the commit message)

File: build/scripts/build-utils.ps1 (hunk 1 above)
Comment (jaredpar) [inline build/scripts/build-utils.ps1]: Then presumably the next command which uses the directory will fail 😄 

I don't meant to be trolly there but there isn't a good answer to this question.  Asking if the directory actually failed to create or already exists is just a variation of the [File.Exists problem](https://blogs.msdn.microsoft.com/jaredpar/2009/12/10/the-file-system-is-unpredictable/).  I could test for existence of the directory after the call but it's a racy check.

File: src/Tools/MicroBuild/run-gitlink.ps1 (hunk 2 above)
Comment (jaredpar) [inline src/Tools/MicroBuild/run-gitlink.ps1]: Sorry, I should have added more detail to this commit about that.  I had put a write up of the problem in our previous convo and didn't think to add it here.  

The PR has a discussion of why this approach was taken now.

**Pull request merged**